### PR TITLE
Fix MSAL Issue #741

### PR DESCRIPTION
**Pull request by shahzaibj**
See #741

### CODE REVIEW

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 1]
```diff
@@ -51,6 +52,7 @@
 import com.microsoft.identity.client.internal.telemetry.DefaultEvent;
 import com.microsoft.identity.client.internal.telemetry.Defaults;
 import com.microsoft.identity.common.adal.internal.tokensharing.TokenShareUtility;
+import com.microsoft.identity.common.exception.ArgumentException;
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: unused import

Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: Addressed

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 2]
```diff
@@ -1292,18 +1295,19 @@ private String getRequestTenantId(@NonNull final PublicClientApplicationConfigur
...
-        }
+        validateNonNullArg(activity, "Activity");
+        validateNonNullArg(scopes, "Scopes");
+        validateNonNullArg(callback, "Callback");
+    }
 
-        return;
+    protected void validateAcquireTokenSilentParameters(AcquireTokenSilentParameters parameters) throws MsalArgumentException {
+        String authority = parameters.getAuthority();
+        validateNonNullArg(authority, "Authority");
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: also validate `account`, `callback` `scopes`

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 3]
```diff
@@ -1418,8 +1424,10 @@ public void run() {
                     );
 
                     ApiDispatcher.submitSilent(silentTokenCommand);
-                } catch (final BaseException exception) {
-                    callback.onError(exception);
+                } catch (final Exception exception) {
+                    // convert exception to BaseException
+                    BaseException baseException = ExceptionAdapter.baseExceptionFromException(exception);
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: nit : final

**Pull request merged**